projects/vdk-core/src/vdk/api/job_input.py

```diff
@@ -361,3 +361,18 @@ def get_execution_properties(self) -> dict:
                 SELECT id, {pa__job_start_ts_expr} FROM mytable
         """
         pass
+
+    @abstractmethod
+    def cancel_job_execution(self) -> None:
```

Comment:
Let's name it "cancel_remaining_steps" ? Since canceling template execution will not cancel the whole job and it may confuse people.